### DISCUSSION TITLE
Question on Phidget IO Command on HUB0000

### DISCUSSION THREAD

**Discussion by bigl00z3** [artisan-roaster-scope/artisan#804] [Q&A] [answered]
I want to make an audible alarm (louder than the PC sound) to alert me when something is happening. 
I bought a [cheap 5V buzzer](https://www.distrelec.ch/fr/grove-buzzer-seeed-studio-107020000/p/30069924?track=true&no-cache=true&marketingPopup=false). 

The idea is to trigger it with the IO Commands `pulse(c,t[,sn])` or `set(c,b[,sn])`

I can't seem to make it work when connecting to my [Hub HUB0000_0](https://www.phidgets.com/?tier=3&catid=2&pcid=1&prodid=643). I have also connected a [PRE1000](https://www.phidgets.com/?tier=3&catid=64&pcid=57&prodid=719) just to check that Artisan is communicating (it does).

My command have been buttons with `pulse(0,200,620188)` and also `set(0,1,620188)` for On and `set(0,0,620188)` for Off. 
But no sound coming out of the buzzer.
![Screenshot 2022-01-30 at 15 31 17](https://user-images.githubusercontent.com/8886922/151704018-61f31725-b9b0-4793-8f5f-c6b7063ad8fa.png)

But when I test it directly from the Phidget control panel it does work (trigger on/off for the Digital out).
Note: For the described tests I have of course always either the Phidget Control Panel or Artisan open. Never the 2 at the same time. So my problem is not from this.
![Screenshot 2022-01-30 at 15 29 05](https://user-images.githubusercontent.com/8886922/151703940-8e32f221-8a71-4734-aace-7b67ea9cd390.png)

I did borrow a [Phidget Interfacekit 8/8/8](https://www.phidgets.com/?tier=3&catid=2&pcid=1&prodid=1035) and it works perfectly with command: `pulse(7,200,595251);sleep(0.25);pulse(7,200,595251).` The buzzer works and I also tested with LEDs.
![Screenshot 2022-01-30 at 15 37 04](https://user-images.githubusercontent.com/8886922/151704207-846236e1-2982-41a6-b000-b4efa111541a.png)

Is it possible to do it with the HUB0000? Or do I need a separate Phidget with a dedicated Digital out?

I have also seen on [the Artisan page](https://artisan-scope.org/devices/phidgets/#42-digital-output)  that there may be the command `pulsehub(p,t[,<sn>])` but this did not seem to work either (getting a "_No match for command [pulsehub], continuing_" error for this one)

Any ideas or recommendation before I order myself an Interface kit 8/8/8/?

Nico

**Comment by MAKOMO** [comment, marked as answer]
To address the HUB ports directly as digital outputs (incl. PWM) you need to use the "PWM Commands"

```outhub / togglehub / pulsehub```

Note those are PWM Commands and not IO Commands as they are essentially setting the PWM of the correspondng port.

Further note that those outputs deliver only 3.3V max and not the 5V as reported in the HUB manual currently under Digital Outputs (inconsistency reported).

Not sure if those 3.3V are enough to drive your buzzer. If you need a full 5V output you could use the OUT1100 VINT module which also supports PWM.

**Reply by bigl00z3** [reply]
Ok works perfectly as PWM and `pulsehub.` My previous attempt with `pulsehub` wasn't working as I was in IO Command.

Lower voltage is still sufficient for the buzzer. It works very nicely now.

Nico
